### ISSUE TITLE
Error when using firstOrCreate() with "use Searchable;"

### ISSUE THREAD

**Issue by TheDailySpank** [teamtnt/laravel-scout-tntsearch-driver#97]
I have a model that I use firstOrCreate() with in order to allocate the model's ID. When "use Searchable;" is enabled in the model, it's getting a null result for the column tntsearch should index (as expected) and throwing an error causing the model to not be saved at the end of the function.

Enabling use Searchable afterwards and then manually indexing works.

firstOrNew(), add data to model, then save(), in my case, is not a possibility since there are a lot of duplicate actions going on and they all use findOrCreate to keep the workload limited to DB lookups.

Is there a way to ignore NULL results or change the behavior of the indexer to only attempt to add the item to the index on save() ?

**Comment by marksparrish** [comment]
firstOrCreate does save the model when the create part is called.  That is how an id is returned.  You could use the toSearchableArray to set the values.

**Comment by TheDailySpank** [comment]
Some excerpts from my code. $crawler is Crawler from symfony and works fine for what I'm using it for. I'm extracting the text from the HTML and putting both into $page_source and saving it to my database. The source HTML is only stored so it doesn't need to re-download during testing so I don't want it included in the full text index.

laravel.log
```
[2017-06-10 22:02:38] local.ERROR: ErrorException: Undefined index: id in /vendor/teamtnt/tntsearch/src/Support/Collection.php:68
```

Spider.php
```
$page_source = new App\Html;

$page_source->page_id = $page->id;
...
$page_source->html = file_get_contents($url);
$page_source->text = $crawler->filter('body')->text();
$page_source->save();
```


App\Html.php
```
namespace App;
use Illuminate\Database\Eloquent\Model;
use Laravel\Scout\Searchable;

class Html extends Model
{

	use Searchable;
	...

	public function toSearchableArray()
	{
	    return [
	        'text' => $this->text,
	    ];
	}
	...
}
```

Thanks in advance.

**Comment by TheDailySpank** [comment]
I searched some more and figured it  out (think I just needed to sleep on it).

The array returned in toSearchableArray needs the id of the row. It's working with the code below. Might be an idea to have this noted in the docs or at least where it says "// Customize array" in the usage example.

```
public function toSearchableArray()
{
    return [
        'id'   => $this->id,
        'text' => $this->text,
    ];
}
```

Related: https://github.com/teamtnt/laravel-scout-tntsearch-driver/issues/92
